apache/superset
# feat(helm): optionally set pod disruption budgets

### SUMMARY
This PRs optionally adds pod disruption budgets against all Superset deployments. By default, they are not enabled to keep the current behavior but we're now able to set PDBs on any deployment if relevant.

This also comes with a slight refactoring on the selector labels for each deployment: they are now defined as named templates in `_helpers.tpl` to mitigate selectors duplication between deployments and their respective related pod disruption budgets.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS
One can set any `supersetXxx.podDisruptionBudget.enabled` as `true` and add either `minAvailable` or `maxUnavailable` variables and test whether the generated manifests look as expected. 
PDB manifests are not generated when the relevant `podDisruptionBudget.enabled` is `false` and generation fails if any of the enabled PDB defines both `minAvailable` and `maxUnavailable` at the same time.

### ADDITIONAL INFORMATION
